lib/ain-evm/src/executor.rs

```diff
@@ -246,6 +249,67 @@ impl<'backend> AinExecutor<'backend> {
         ))
     }
 
+    pub fn exec_with_tracer(
+        &mut self,
+        signed_tx: &SignedTx,
+    ) -> Result<(Vec<ExecutionStep>, bool, Vec<u8>, u64)> {
+        let to = signed_tx.to().ok_or(format_err!(
+            "debug_traceTransaction does not support contract creation transactions",
+        ))?;
+        self.backend.update_vicinity_from_tx(signed_tx)?;
```

Comment:
Could we lift this so we can remove the `&mut self` and rename it to `call_with_tracer`. This is replaying a TX and should be a read-only call.

Reply:
Makes sense.

Reply:
Resolved in 35e1f6caa6